File: codegen/results/v001/gemma4_26b_nothink/20260407_155240/python/003_fibonacci/solution.py
import sys

def print_fibonacci_up_to_n(n):
    """
    Prints Fibonacci numbers starting from 1, 1, 2... 
    up to the largest Fibonacci number <= n.
    """
    if n < 1:
        return

    a, b = 1, 1
    
    # The sequence starts with 1, 1. We handle the first two 1s explicitly
    # or via a loop logic that checks the current number against n.
    while a <= n:
        print(a)
        # Calculate next Fibonacci number
        a, b = b, a + b

def main():
    # Check if exactly one argument is provided (sys.argv[0] is the script name)
    if len(sys.argv) != 2:
        return

    try:
        # Parse the command line argument as an integer
        n = int(sys.argv[1])
        print_fibonacci_up_to_n(n)
    except ValueError:
        # If the argument is not a valid integer, we exit silently 
        # or handle error as per requirements.
        pass

if __name__ == "__main__":
    main()

File: codegen/results/v001/gemma4_26b_nothink/20260407_155240/go/003_fibonacci/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	// Check if an argument was provided
	if len(os.Args) < 2 {
		return
	}

	// Parse the input argument N
	nStr := os.Args[1]
	n, err := strconv.Atoi(nStr)
	if err != nil {
		// If the input is not a valid integer, we exit silently 
		// (or you could handle error specifically)
		return
	}

	// If N < 1, the requirement says print nothing.
	if n < 1 {
		return
	}

	// Initialize the first two numbers of the Fibonacci sequence
	a := 1
	b := 1

	// Print the first number
	fmt.Println(a)

	// If the second number is also <= N, we enter the loop
	// Note: The sequence starts 1, 1, 2...
	// We handle the second '1' separately to manage the loop logic correctly
	if b <= n {
		fmt.Println(b)
	} else {
		// This case handles if N was 1, but the second 1 is already handled.
		// However, with N >= 1, the loop below handles subsequent numbers.
		return
	}

	// Calculate and print subsequent Fibonacci numbers
	for {
		next := a + b
		if next > n {
			break
		}
		fmt.Println(next)
		// Update values for the next iteration
		a = b
		b = next
	}
}